modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/future/HullWhiteOneFactorPiecewiseConstantConvexityFactorProvider.java

```diff
@@ -0,0 +1,421 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.rate.future;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.pricer.impl.rate.model.HullWhiteOneFactorPiecewiseConstantInterestRateModel;
+import com.opengamma.strata.pricer.impl.rate.model.HullWhiteOneFactorPiecewiseConstantParameters;
+
+/**
+ * Provider of convexity factor for futures based on Hull-White one factor model with piecewise constant volatility.
+ * <p>
+ * Reference: Henrard, M. "The Irony in the derivatives discounting Part II: the crisis", Wilmott Journal, 2010, 2, 301-316
+ */
+@BeanDefinition(builderScope = "private")
+public final class HullWhiteOneFactorPiecewiseConstantConvexityFactorProvider implements ImmutableBean, Serializable {
```

Comment:
Should we have a provider of only the "convexity adjustment" or more generally of the "Hull-White parameters". I'm more in favor of the second. The same provider should be used for any instrument and any valuation.
Same question will appear for CMS: do we use the "SabrVolatilitySwaptionProvider" or do we create another provider. I'm more of the opinion that the provider provides the data/parameters and it is up to the pricer to use them wisely. In the present case, I would prefer to have a "HullWhiteOneFactorPiecewiseConstantParametersProvider".